modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/Blog.cs

```diff
@@ -0,0 +1,41 @@
+﻿using JetBrains.Annotations;
+using System;
+using Volo.Abp;
+using Volo.Abp.Domain.Entities.Auditing;
+using Volo.Abp.MultiTenancy;
+using Volo.CmsKit.Blogs.Extensions;
+
+namespace Volo.CmsKit.Blogs
+{
+    public class Blog : FullAuditedAggregateRoot<Guid>, IMultiTenant
+    {
+        public Blog(
+            Guid id,
+            [NotNull] string name,
+            [NotNull] string urlSlug,
+            [CanBeNull] Guid? tenantId = null) : base(id)
+        {
+            SetName(name);
+            SetUrlSlug(urlSlug);
+            TenantId = tenantId;
+        }
+
+        public string Name { get; protected set; }
+
+        public string UrlSlug { get; protected set; }
+
+        public Guid? TenantId { get; }
+
+        public void SetName(string name)
+        {
+            Name = Check.NotNullOrWhiteSpace(name, nameof(name), maxLength: BlogConsts.MaxNameLength);
+        }
+
+        public void SetUrlSlug(string urlSlug)
+        {
+            Check.NotNullOrWhiteSpace(urlSlug, nameof(urlSlug), maxLength: BlogConsts.MaxNameLength);
+
+            UrlSlug = urlSlug.NormalizeAsUrlSlug();
```

Comment:
Why do we need to generate the slug on the backend? We can generate the slug on the client-side to help the user but I don't think we should change the slug on the backend.

Reply:
Also, Client-side has the same logic, but still custom http request can be sent and backend secures itself on this situation. For example, if a request is sent with special characters such as `&` or `?`, routing will be broken. To keep routes working, backend normalizes always.

Reply:
I'm not sure we should change the given slug. The caller should be responsible for that. Maybe we can make the slug optional and generate the slug on the backend if not specified. What do you think?

Reply:
Generating automatically, when only slug is empty looks OK for me. 

In other way, client will be responsible of wrong slugs.

Reply:
Might an exception be thrown if url is invalid. How about that? 

Keeping unresolvable URL in database doesn't look good for me @ilkayilknur

Reply:
We can use https://github.com/ctolkien/Slugify to generate slugs on the backend. We can use this library to normalize the slugs as well. @enisn